### PR TITLE
chore: performance improvements for account derivation

### PR DESCRIPTION
**Pull request by fedeerbes**
# 🔘 PR Type

- [ ] Bugfix
- [X] Enhancement
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

# 📜 Background
This PR adds to new methods to optimize the performance of restoration of account from the seed phrase. It exposes one new method `restoreWalletWithAccountsMobile` which uses `getActiveAccountsFromRootNode` and `getWalletFromRootNode`. the idea is to get the root node once and use it where is necessary since the calculation is expensive in terms of CPU. 
It also adds the method `waitFrame`, a simple wrapper of requestAnimationFrame to let the CPU breathe between expensive operations 

Issue Link: #[ENG-2449](https://linear.app/xverseapp/issue/ENG-2449/performance-improvements-for-account-derivation)

# 🔄 Changes

- [ ] Yes, Incompatible API changes
- [x] No, Adds functionality (backwards compatible)
- [ ] No, Bug fixes (backwards compatible)

Changes:
- Modify restoreWalletWithAccounts to generate the root node and uses a for loop instead of promise all to improve performance
- Create getActiveAccountsFromRootNode which receives the root node instead of generate it
- fetchActiveAccounts now uses getActiveAccountsFromRootNode to not introduce a breaking change
- Create getWalletFromRootNode which receives the root node instead of generate it
- walletFromSeedPhrase now uses getWalletFromRootNode to not introduce a breaking change
- Unify accountType definition
- None of this changes are breaking change

Impact:
- Improves performance for the mobile app when restoring accounts
- Impacts on mobile app and web extensions

# 🧪 Testing
We should test mobile and web extension in this PR and the following flows:
- Create account 
- Create wallet
- Restore wallet
- Restore accounts
- Change network on settings

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (m-aboelenein) [inline vite.config.ts:1]: why did we need to add this ?

File: wallet/index.ts [hunk 2]
```diff
@@ -71,40 +70,46 @@ export async function newWallet(): Promise<BaseWallet> {
...
 
-  const master = bip32.fromSeed(seed);
-  const masterPubKey = master.publicKey.toString('hex');
   const stxPublicKey = publicKeyToString(getPublicKey(createStacksPrivateKey(privateKey)));
 
   // derive segwit btc address
+  await waitFrame();
   const btcChild = master.derivePath(getBitcoinDerivationPath({ index, network }));
+
+  await waitFrame();
```
Comment (m-aboelenein) [inline wallet/index.ts:101]: i am not sure about this one did it make a notable difference in performance?

File: vite.config.ts (hunk 1 above)
Comment (fedeerbes) [reply vite.config.ts:1]: cause it throwing an eslint error which doesn't make sense 

`'vitest' should be listed in the project's dependencies, not devDependencies.`

File: wallet/index.ts (hunk 2 above)
Comment (fedeerbes) [reply wallet/index.ts:101]: in mobile the difference is noticeable. we gain several frames between tasks with this

**Review by dhriaznov — APPROVED** [review]
LGTM

**Comment by dhriaznov** [comment]
There's a git conflict to solve @fedeerbes

File: wallet/index.ts (hunk 2 above)
Comment (fedeerbes) [reply wallet/index.ts]: removed

File: vite.config.ts (hunk 1 above)
Comment (fedeerbes) [reply vite.config.ts]: removed, not needed anymore

**Review by m-aboelenein — APPROVED** [review]

File: account/index.ts [hunk 3]
```diff
@@ -43,11 +39,26 @@ export const fetchActiveAccounts = async (
   });
 };
 
+export const fetchActiveAccounts = async (
```
Comment (yknl) [inline account/index.ts:42]: Does this function become obsolete after the seed vault PR https://github.com/secretkeylabs/xverse-core/pull/169?

Comment (fedeerbes) [reply account/index.ts:42]: i have to take a look at that PR and get the context, maybe @m-aboelenein can have a quicker reply

Comment (m-aboelenein) [reply account/index.ts:42]: this is not related to the seed vault yet it only handles storage of the seed phrase

**Pull request merged**